src/API/Endpoints/Reports/TotalRefunds.php

```diff
@@ -0,0 +1,187 @@
+
+		return new \WP_REST_Response(
+			[
+				'data' => $data,
+			]
+		);
+	}
+
+	public function get_data( $start, $end, $intervalStr ) {
+
+		$this->payments = $this->get_payments( $start->format( 'Y-m-d H:i:s' ), $end->format( 'Y-m-d H:i:s' ) );
+
+		$tooltips = [];
+		$refunds  = [];
+
+		$interval = new \DateInterval( $intervalStr );
+
+		$periodStart = clone $start;
+		$periodEnd   = clone $start;
+
+		// Subtract interval to set up period start
+		date_sub( $periodStart, $interval );
+
+		while ( $periodStart < $end ) {
+
+			$refundsForPeriod = $this->get_refunds( $periodStart->format( 'Y-m-d H:i:s' ), $periodEnd->format( 'Y-m-d H:i:s' ) );
+
+			switch ( $intervalStr ) {
+				case 'PT12H':
+					$periodLabel = $periodStart->format( 'D ga' ) . ' - ' . $periodEnd->format( 'D ga' );
+					break;
+				case 'PT3H':
+					$periodLabel = $periodStart->format( 'D ga' ) . ' - ' . $periodEnd->format( 'D ga' );
+					break;
+				case 'PT1H':
+					$periodLabel = $periodStart->format( 'D ga' ) . ' - ' . $periodEnd->format( 'D ga' );
+					break;
+				default:
+					$periodLabel = $periodStart->format( 'M j, Y' ) . ' - ' . $periodEnd->format( 'M j, Y' );
+			}
+
+			$refunds[] = [
+				'x' => $periodEnd->format( 'Y-m-d H:i:s' ),
+				'y' => $refundsForPeriod,
+			];
+
+			$tooltips[] = [
+				'title'  => $refundsForPeriod . ' ' . __( 'Refunds', 'give' ),
+				'body'   => __( 'Total Refunds', 'give' ),
+				'footer' => $periodLabel,
+			];
+
+			// Add interval to set up next period
+			date_add( $periodStart, $interval );
+			date_add( $periodEnd, $interval );
+		}
+
+		$totalRefundsForPeriod = $this->get_refunds( $start->format( 'Y-m-d H:i:s' ), $end->format( 'Y-m-d H:i:s' ) );
+		$trend                 = $this->get_trend( $start, $end, $refunds );
+
+		// Create data objec to be returned, with 'highlights' object containing total and average figures to display
+		$data = [
+			'datasets' => [
+				[
+					'data'      => $refunds,
+					'tooltips'  => $tooltips,
+					'trend'     => $trend,
+					'highlight' => $totalRefundsForPeriod,
+				],
+			],
+		];
+
+		return $data;
+
+	}
+
+	public function get_trend( $start, $end, $income ) {
+
+		$interval = $start->diff( $end );
+
+		$prevStart = clone $start;
+		$prevStart = date_sub( $prevStart, $interval );
+
+		$prevEnd = clone $start;
+
+		$prevRefunds    = $this->get_prev_refunds( $prevStart->format( 'Y-m-d H:i:s' ), $prevEnd->format( 'Y-m-d H:i:s' ) );
+		$currentRefunds = $this->get_refunds( $start->format( 'Y-m-d H:i:s' ), $end->format( 'Y-m-d H:i:s' ) );
+
+		$trend = 0;
+		if ( $prevRefunds > 0 && $currentRefunds > 0 ) {
+			$trend = round( ( ( ( $prevRefunds - $currentRefunds ) / $currentRefunds ) * 100 ), 1 );
+		}
+
+		return $trend;
+	}
+
+	public function get_refunds( $startStr, $endStr ) {
+
+		$refunds = 0;
+		foreach ( $this->payments as $payment ) {
+			if ( $payment->status == 'refunded' && $payment->date > $startStr && $payment->date < $endStr ) {
```

Comment:
@henryholtgeerts Is Date string formatted in other than UNIX timestamp? I think date string comparison with `<` & `>` will not give you the desired result.

Reply:
Because $dateStr is in the same format as $payment->date, php is able to compare them, see: https://www.geeksforgeeks.org/comparing-two-dates-in-php/